website/docs/recipes/kafka.md

```diff
@@ -169,3 +170,217 @@ private MessageAndMetadata createPactRepresentationFor(Map<String, Object> metad
+}
+
+private KafkaJsonSchemaDeserializer<ConsumerDomainRecord> getProductionKafkaDeserializer() {
+    Map<String, Object> props = Map.of(
+            KafkaJsonSchemaDeserializerConfig.SCHEMA_REGISTRY_URL_CONFIG, "mock://anything",
+            KafkaJsonSchemaDeserializerConfig.JSON_VALUE_TYPE, ConsumerDomainRecord.class.getName()
+    );
+    KafkaJsonSchemaDeserializer<ConsumerDomainRecord> deserializer = new KafkaJsonSchemaDeserializer<>(schemaRegistryClient);
+    deserializer.configure(props, IS_KEY);
+    tryRegisterSchemaForValue();
+    return deserializer;
+}
+```
+Those last two methods are worthy of note. The getProductionKafkaDeserializer method is intended to reach into your production code (otherwise we're just testing the test code, and not a lot of point in that!) and get hold of the deserializer used. Most Kafka consumer applications would use out-of-the-box deserializers, which are simply a configuration parameter for the KafkaConsumer. You may want to reach into the config and pull out the class name defined for the Deserializer, and then instantiate it perhaps using reflection - I'm going to leave that as an exercise for you as I don't want to play with your production code here! The point is that the implementation given here needs to be replaced with some production code for your test to have much meaning.
+
+Also keep an eye out for the config for the URL to the Schema Registry. As this won't be running for our tests, we configure the value of the URL with "mock://" - this tells the deserializer not to try and communicate to the registry.
+
+Moving on, let's complete the last few private methods:
+```
+private void tryRegisterSchemaForValue() {
+    try {
+        registerSchemaForValue();
+    } catch (IOException | RestClientException e) {
+        e.printStackTrace();
+    }
+}
+
+private int registerSchemaForValue() throws IOException, RestClientException {
+    return schemaRegistryClient.register(TOPIC_NAME + "-value", getConsumerDomainRecordSchema());
+}
+
+private JsonSchema getConsumerDomainRecordSchema() throws IOException {
+    return JsonSchemaUtils.getSchema(
+            new ConsumerDomainRecord("name"),
+            SpecificationVersion.DRAFT_7,
+            USE_ONEOF_FOR_NULLABLES,
+            FAIL_UNKNOWN_PROPERTIES,
+            new ObjectMapper(),
+            schemaRegistryClient
+    );
+}
+```
+We're using a MockSchemaRegistryClient which is provided out-of-the-box. Given that we're not actually running Kafka, we're also not running the Schema Registry, so we don't want anything to try and communicate with it, and thats what this class does for us. We need to register the subject (usually the Kafka topic + either "-key" or "-value") along with the JsonSchema itself into the MockSchemaRegistryClient.
+
+## The Pact
+As ever, running the consumer test will generate the Pact file (JSON), and push it out to the Pact Broker (well, you are using one, right?!):
```

Comment:
push it out to the Pact Broker if running in CI, or publishPacts flag is set?

Is it worth making this distinction, as generally you wouldn't publish pacts from local runs?

Reply:
Yes, indeed... will fix!